cognite/client/_api/assets.py

```diff
@@ -59,17 +59,54 @@ def __iter__(self) -> Generator[Asset, None, None]:
         """
         return self.__call__()
 
+    @overload
     def retrieve(
         self, id: Union[int, List[int]] = None, external_id: Union[str, List[str]] = None
```

Comment:
Must be Union[int, List[int], None]. Anything that can be None must have type None, Optional[T] or Union[T1, T2, None]